test/config_file.c

```diff
@@ -269,6 +269,79 @@ mountprefix=/mnt/myrauc/\n";
 	g_clear_error(&ierror);
 }
 
+static void config_file_boot_attempts(ConfigFileFixture *fixture,
+		gconstpointer user_data)
+{
+	RaucConfig *config;
+	GError *ierror = NULL;
+	gchar* pathname;
+
+	const gchar *valid_boot_attempts_cfg_file = "\
+[system]\n\
+compatible=FooCorp Super BarBazzer\n\
+bootloader=uboot\n\
+boot-attempts=5\n\
+boot-attempts-primary=10\n\
+mountprefix=/mnt/myrauc/\n";
+	const gchar *invalid_boot_attempts_cfg_file = "\
+[system]\n\
+compatible=FooCorp Super BarBazzer\n\
+bootloader=uboot\n\
+boot-attempts=urks\n\
+boot-attempts-primary=10\n\
+mountprefix=/mnt/myrauc/\n";
+	const gchar *boot_attempts_invalid_bootloader_cfg_file = "\
+[system]\n\
+compatible=FooCorp Super BarBazzer\n\
+bootloader=grub\n\
+boot-attempts=5\n\
+boot-attempts-primary=10\n\
+mountprefix=/mnt/myrauc/\n";
+	const gchar *negative_boot_attempts_cfg_file = "\
+[system]\n\
+compatible=FooCorp Super BarBazzer\n\
+bootloader=uboot\n\
+boot-attempts=5\n\
+boot-attempts-primary=-10\n\
+mountprefix=/mnt/myrauc/\n";
+
+
+	pathname = write_tmp_file(fixture->tmpdir, "valid_bootloader.conf", valid_boot_attempts_cfg_file, NULL);
+	g_assert_nonnull(pathname);
+
+	g_assert_true(load_config(pathname, &config, &ierror));
+	g_assert_no_error(ierror);
+
+	g_free(pathname);
+
+	pathname = write_tmp_file(fixture->tmpdir, "invalid_bootloader.conf", invalid_boot_attempts_cfg_file, NULL);
+	g_assert_nonnull(pathname);
+
+	g_assert_false(load_config(pathname, &config, &ierror));
+	g_assert_cmpstr(ierror->message, ==, "Key file contains key \342\200\234boot-attempts\342\200\235 in group \342\200\234system\342\200\235 which has a value that cannot be interpreted.");
```

Comment:
This might depend too much on glib implementation details. Perhaps use g_assert_error with domain and code? Or check only that specific substrings are in the message?

Reply:
Mh, yes, might be better here indeed